### PR TITLE
Tweaks to `LDT/DeVeny`, primarily wavelength-related

### PR DESCRIPTION
**Pull request by tbowers7**
This is an update to the `LDT/DeVeny` spectrograph class primarily aimed at wavelength calibration.

Major changes include:
* Addition of canonical 150L, 300L, 600L, and 1200L wavelength templates (and removal of previous ones);
* Addition of the wavelength-template generator script;
* Adjustments to the grating-specific wavelength solution polynomial orders;
* Use of the new `cenwave` meta card (introduced in #1384) to describe the grating tilt; and
* Addition of 4 new blue lines (<3350Å) to the wavelength lists (1 `CdI`, and 3 `HgI`).

Minor changes include:
* Expansion of the `HgI` and `CdI` NIST table wavelength coverage;
* Addition of `DeVeny`-specific line lists for `HgI` and `CdI` for the 1200L grating that include faint lines that are not generally applicable; and
* Correction of the effective aperture of the LDT.

Further discussion:
The four blue arc lines are:
```
HgI |  2968.1495
HgI |  3022.384
CdI |  3134.0746
HgI |  3342.4450
```
This is the `QA` plot for `DeVeny`'s 300L grating from which the lines were identified:
Arc_1dfit_A_1_DET01_S0248

Unit Tests Pass:
```
============================= test session starts ==============================
collected 329 items                                                            
======================= 329 passed in 1971.62s (0:32:51) =======================
```

### CODE REVIEW

File: pypeit/core/wavecal/spectrographs/templ_ldt_deveny.py [hunk 1]
```diff
@@ -0,0 +1,146 @@
+""" Generate the wavelength templates for LDT/DeVeny"""
```
Comment (profxj) [inline pypeit/core/wavecal/spectrographs/templ_ldt_deveny.py:1]: Thanks for this!

File: pypeit/spectrographs/ldt_deveny.py [hunk 2]
```diff
@@ -113,6 +113,7 @@ def init_meta(self):
         # Extras for config and frametyping
         self.meta['idname'] = dict(ext=0, card='IMAGETYP')
         self.meta['dispangle'] = dict(ext=0, card='GRANGLE', rtol=1e-3)
+        self.meta['cenwave'] = dict(card=None, compound=True, rtol=5)
```
Comment (profxj) [inline pypeit/spectrographs/ldt_deveny.py]: Does 5 here mean 5Ang?  Seems a touch big if so
unless the resolution is very low..

**Review by profxj — APPROVED** [review]
Great stuff.  I am approving, but please don't merge
until we have new DevSuite tests in place for the
new gratings.

Thx!

File: pypeit/spectrographs/ldt_deveny.py (hunk 2 above)
Comment (tbowers7) [reply pypeit/spectrographs/ldt_deveny.py]: Our motorized grating tilt control is accurate to 0.01º.  For the 300g/mm grating, this translates to a 10Å precision, for 600g/mm, a 5Å precision.  I can lower the matching value (the new `Keck/LRIS` card has `rtol=2.0`), but I'm not sure it'll make much practical difference.

File: pypeit/core/wavecal/spectrographs/templ_ldt_deveny.py (hunk 1 above)
Comment (tbowers7) [reply pypeit/core/wavecal/spectrographs/templ_ldt_deveny.py:1]: Input templates being included in pypeit/PypeIt-development-suite#203.

**Comment by tbowers7** [comment]
`dev-suite` tests for updated (and new) setups all pass:
```
Running tests on the following instruments:
    ldt_deveny

Reducing data from ldt_deveny for the following setups:
    DV1
    DV2
    DV5
    DV6
    DV8
    DV9

--- PYPEIT DEVELOPMENT SUITE PASSED 6/6 TESTS (Masters ignored) ---
Testing Started at 2022-05-05T09:25:27.287120
Testing Completed at 2022-05-05T09:53:05.031628
Total Time: 0:27:37.744508
```

File: pypeit/spectrographs/ldt_deveny.py [hunk 3]
```diff
@@ -170,6 +169,25 @@ def compound_meta(self, headarr, meta_key):
...
+            # Grating angle in radians
+            theta = np.deg2rad(float(headarr[0]['GRANGLE']))
+            # Wavelength in Angstroms
+            wavelen = (
+                (np.sin(COLL + theta) + np.sin(COLL + theta - CAMCOL))  # Angles
+                * 1.0e7                                                 # Angstroms/mm
+                / float(headarr[0]["GRATING"].split("/")[0])            # lines / mm
+            )
+            # Round the wavelength to the nearest 5A
+            return np.around(wavelen / 5, decimals=0) * 5
```
Comment (kbwestfall) [inline pypeit/spectrographs/ldt_deveny.py:196]: Nice!

**Review by kbwestfall — APPROVED** [review]
Looks good to me!

**Comment by tbowers7** [comment]
Hmm, CI failing... looking into it...

**Review by profxj — APPROVED** [review]
Approving

When things merge, tickle me to run a `develop` test just to be sure
things went smoothly

**Comment by tbowers7** [comment]
**Note:** The CI failures are being caused by a corrupted `IERS_A` file downloaded from USNO via AstroPy (v5.0.4).  This appears to be an issue with the USNO server because using v5.0.2 (NASA server for the `IERS_A` file) passes these tests.

AstroPy is aware of the issue and is working it.  astropy/astropy#13223

**Pull request merged**